core/src/test/java/org/apache/calcite/test/RelMetadataTest.java

```diff
@@ -3639,6 +3642,37 @@ public void checkAllPredicatesAndTableSetOp(String sql) {
         is(mq.getPopulationSize(rel, bitSetOf(0))));
   }
 
+  /**
+   * Test that RelMdPopulationSize is calculated based on the RelMetadataQuery#getRowCount().
+   *
+   * @see <a href="https://issues.apache.org/jira/browse/CALCITE-5647">[CALCITE-5647]</a>
+   */
+  @Test public void testPopulationSizeFromValues() {
+    // Arrange
+    final String sql = "values(1,2,3)";
+    final RelNode rel = sql(sql).toRel();
+    assertThat(rel, instanceOf(Values.class));
+
+    RelMetadataProvider provider = RelMdPopulationSize.SOURCE;
+
+    List<MetadataHandler<?>> handlers =
+        provider.handlers(BuiltInMetadata.PopulationSize.Handler.class);
+
+    // Act
```

Comment:
I am not sure that these 3 comments are useful.
In fact, this one confused me.

Reply:
This is a pretty common way of documenting tests:
https://automationpanda.com/2020/07/07/arrange-act-assert-a-pattern-for-writing-good-tests/

- Arrange would be setting up the test (that isn't global setup like the @Setup function)
- Act would be executing the test behavior
- Assert would be validating the result.

I can remove these comments but don't really see the harm in them.

Reply:
Comments should provide information that is not clear from the code. "Assert" for example does not. And "act" could mean many thing. I think it would be better without them.

But here the test is not showing that you fixed a bug, it is showing just that a specific method is called. The bug should manifest itself in producing an incorrect or suboptimal result for some code transformation. That would be a convincing test.